src/server/dfly_main.cc

```diff
@@ -327,16 +331,26 @@ bool RunEngine(ProactorPool* pool, AcceptServer* acceptor) {
     Listener* uds_listener = new Listener{Protocol::REDIS, &service};
     error_code ec = acceptor->AddUDSListener(unix_sock.c_str(), uds_listener);
     if (ec) {
-      LOG(WARNING) << "Could not open unix socket " << unix_sock << ", error " << ec;
+      if (tcp_disabled) {
+        LOG(ERROR) << "Could not open unix socket " << unix_sock
+                   << ", and TCP listening is disabled (error: " << ec << "). Exiting.";
+        exit(1);
+      } else {
+        LOG(WARNING) << "Could not open unix socket " << unix_sock << ", error " << ec;
+      }
       delete uds_listener;
     } else {
       LOG(INFO) << "Listening on unix socket " << unix_sock;
       unlink_uds = true;
     }
+  } else if (tcp_disabled) {
+    LOG(ERROR)
+        << "Did not receive a unix socket to listen to, yet TCP listening is disabled. Exiting.";
+    exit(1);
   }
 
   std::uint16_t admin_port = GetFlag(FLAGS_admin_port);
-  if (admin_port != 0) {
+  if (admin_port != 0 && !tcp_disabled) {
```

Comment:
I think it's fine to let `admin_port` allow TCP connectivity independently from `--port`. This allows maximum flexibility